Automattic/jetpack
# Jetpack AI: fix first logo notice

## Proposed changes:
This PR addresses an edge case discovered while working the first logo generation feature. If the user has oddly gone over the fair usage limit, we'd just try to generate and get an error.
With this change, we check and show a proper fair usage notice instead.

Also, while reviewing the conditions we use, the PR updates the type definitions and constants to use the new fair usage limit.

### Other information:


## Jetpack product discussion
p1728486772436309/1728484828.652839-slack-C054LN8RNVA

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
Use a fresh site (or delete logo history from the devtools application tab, localStorage, look for `logo-history-...` entry). 

Go to the editor and insert a logo block. Use the AI toolbar button to open the modal.

If the site has a title and description, a request for a logo generation should trigger.

Now either start with a new site or delete the logo history, mock a user gone over fair usage limit, sandbox the public api and use these filters:

```php
add_filter( 'jetpack_ai_current_period_requests_count', function() { return 3200; } );
add_filter( 'jetpack_ai_all_time_requests_count', function() { return 4300; } );
add_filter( 'jetpack_ai_tier_licensed_quantity', function() { return 1; } );
```
